examples/dynamic_fanout_container.py

```diff
@@ -5,27 +5,22 @@
 More details can be found here: https://github.com/argoproj-labs/hera-workflows/issues/250
 """
 
-from hera import InputFrom, Task, Workflow, WorkflowService
+from hera import Task, Workflow
 
-
-with Workflow(
-    "dynamic-fanout-container", service=WorkflowService(host="https://my-argo-server.com", token="my-auth-token"),
-) as w:
-
-    # this can be anything! e.g fetch from some API, then in parallel process all entities; chunk database records
+# assumes you used `hera.set_global_token` and `hera.set_global_host` so that the workflow can be submitted
+with Workflow("dynamic-fanout-container") as w:
+    # this can be anything! e.g. fetch from some API, then in parallel process all entities; chunk database records
     # and process them in parallel, etc.
     generate_task = Task(
         "generate", image="alpine:latest", command=["echo", '[{"value": "a"}, {"value": "b"}, {"value": "c"}]'],
     )
-
     fanout_task = Task(
         "fanout",
-        input_from=InputFrom("generate", ["value"]),
+        with_param=generate_task.get_result(),  # this make the task fan out over the `with_param`
+        inputs=[generate_task.get_result_as('value')],  # this sets the correct input parameter so the result is used
         image="alpine:latest",
-        command=["echo"],
-        args=["{{inputs.parameters.value}}"],
+        command=["echo", "{{inputs.parameters.value}}"],
     )
-
```

Comment:
This will make the `with_param` redundant because every task is getting the entire output of `generate_task`

We need to do this:

```python
    fanout_task = Task(
        "fanout",
        with_param=generate_task.get_result(),  # this make the task fan out over the `with_param`
        inputs=[Parameter(name="value", value="{{item.value}}")],  # this sets the correct input parameter so the result is used
        image="alpine:latest",
        command=["echo", "{{inputs.parameters.value}}"],
    )
```
Since this isn't a python function it's not as easy to deduce the argo-templates (maybe in the future?). However, this could serve as a good example of how people can use the native template system.

Reply:
An alternative workflow is:

```python
from hera import Task, Workflow

def echo(value: str) -> str:
    return f"echo {value}"

with Workflow(
    "dynamic-fanout-container",
) as w:

    # this can be anything! e.g fetch from some API, then in parallel process all entities; chunk database records
    # and process them in parallel, etc.
    generate_task = Task(
        "generate",
        image="alpine:latest",
        command=["echo", '[{"value": "a"}, {"value": "b"}, {"value": "c"}]'],
    )

    fanout_task = Task(
        "fanout",
        echo,
        with_param=generate_task.get_result(),  # this make the task fan out over the `with_param`
        image="alpine:latest",
        command=["sh"]
    )

    generate_task >> fanout_task
```

However, this will print `{value:x}` for each step. This is because it's impossible to know what the content of the `with_param` will be (could be a list of values, or a list of dicts). Hera will only assume the `with_param` contains dicts if the consuming function has multiple arguments.

If we'd use this example, it'd be better if `generate_task` produced a list instead.

Reply:
Adjusted based on the first suggestion!